src/main/java/tool/TrivyGrypeOutputProcessor.java

```diff
@@ -0,0 +1,213 @@
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pique.model.Diagnostic;
+import pique.model.Finding;
+import pique.utility.PiqueProperties;
+import toolOutputObjects.RelevantVulnerabilityData;
+import utilities.helperFunctions;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TrivyGrypeOutputProcessor implements IOutputProcessor<RelevantVulnerabilityData> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TrivyGrypeOutputProcessor.class);
+
+    /**
+     * Gets the vulnerabilities array from the complete Tool results
+     *
+     * @param results tool output in String format
+     * @return a jsonArray of Tool results or null if no vulnerabilities are found
+     */
+    @Override
+    public JSONArray getVulnerabilitiesFromToolOutput(String results, String toolName) {
+        try {
+            if (toolName.contains("Grype")) {
+                JSONArray matches = new JSONObject(results).optJSONArray("matches");
+                JSONArray vulnerabilities = new JSONArray();
+                for (int i = 0; matches != null && i < matches.length(); i++) {
+                    JSONObject vulnerability = matches.getJSONObject(i).optJSONObject("vulnerability");
+                    if (vulnerability != null) vulnerabilities.put(vulnerability);
+                }
+                return vulnerabilities;
+            }
+            else if (toolName.contains("Trivy")) {
+                JSONArray resultsArray = new JSONObject(results).optJSONArray("Results");
+                JSONArray vulnerabilities = new JSONArray();
+                if (resultsArray != null && resultsArray.length() > 0) {
+                    JSONArray vulnerabilitiesArray = resultsArray.optJSONObject(0).optJSONArray("Vulnerabilities");
+                    for (int i = 0; vulnerabilitiesArray != null && i < vulnerabilitiesArray.length(); i++) {
+                        JSONObject vulnerability = vulnerabilitiesArray.optJSONObject(i);
+                        if (vulnerability != null) vulnerabilities.put(vulnerability);
+                    }
+                }
+                return vulnerabilities;
+            }
+        }
+        catch (JSONException e) {
+            LOGGER.warn("Unable to read results from {} output", toolName);
+        }
+
+        return null;
+    }
+
+    /**
+     * Processes the JSONArray of tool output vulnerabilities into java objects.
+     * This makes it easier to store and process data into Findings later.
+     * Once created, these objects are read-only.
+     *
+     * @param jsonVulns JSONArray of tool output vulnerabilities
+     * @return ArrayList of RelevantVulnerabilityData java objects
+     */
+    @Override
+    public ArrayList<RelevantVulnerabilityData> processToolVulnerabilities(JSONArray jsonVulns, String toolName) {
```

Comment:
In this method (or potentially even the whole class), you might consider using the GSON library to translate json to java objects.  I see a lot of commented-out code that suggests the json parsing logic might be troublesome.  Working with java objects is generally easier.  If your json parsing is working easily, feel free to disregard this comment.